mapie/metrics.py

```diff
@@ -307,3 +308,397 @@ def top_label_ece(
+        size of sets (n_classes +1) is computed.
+
+    Returns
+    -------
+    NDArray of shape (n_alpha,)
+
+    Examples
+    --------
+    >>> from mapie.metrics import classification_ssc_score
+    >>> import numpy as np
+    >>> y_true = y_true_class = np.array([3, 3, 1,])
+    >>> y_pred_set = np.array([
+    ... [False, False, True, True],
+    ... [False, True, False, True],
+    ... [False, True, True, False]
+    ... ])
+    >>> print(classification_ssc_score(y_true,y_pred_set,num_bins=2))
+    [1.]
+    """
+    return np.min(classification_ssc(y_true, y_pred_set, num_bins), axis=1)
+
+
+def _gaussian_kernel(
+    x: NDArray,
+    kernel_size: int
+) -> NDArray:
+    """
+    Computes the gaussian kernel of x. (Used in hsic function)
+
+    Parameters
+    ----------
+    x: NDArray
+        The values from which to compute the gaussian kernel.
+    kernel_size: int
+        The variance (sigma), this coefficient controls the width of the curve.
+    """
+    norm_x = x ** 2
+    dist = -2 * np.matmul(x, x.transpose((0, 2, 1))) \
+        + norm_x + norm_x.transpose((0, 2, 1))
+    return np.exp(-dist / kernel_size)
+
+
+def hsic(
+    y_true: NDArray,
+    y_intervals: NDArray,
+    kernel_sizes: ArrayLike = (1, 1)
+) -> NDArray:
+    """
+    Compute the square root of the hsic coefficient. HSIC is Hilbert-Schmidt
+    independence criterion that is a correlation measure. Here we use it as
+    proposed in [4], to compute the correlation between the indicator of
+    coverage and the interval size.
+
+    If hsic is 0, the two variables (he indicator of coverage and the
+    interval size) are independant.
+
+    [4] Feldman, S., Bates, S., & Romano, Y. (2021).
+    Improving conditional coverage via orthogonal quantile regression.
+    Advances in Neural Information Processing Systems, 34, 2060-2071.
+
+    Parameters
+    ----------
+    y_true: NDArray of shape (n_samples,)
+        True labels.
+    y_intervals: NDArray of shape (n_samples, 2, n_alpha) or (n_samples, 2)
+        Prediction sets given by booleans of labels.
+    kernel_sizes: int
+        The variance (sigma), this coefficient controls the width of the curve.
+
+    Returns
+    -------
+        NDArray of shape (n_alpha,)
+    One hsic correlation coefficient by alpha.
+
+    Examples
+    -------
+    >>> from mapie.metrics import hsic
+    >>> import numpy as np
+    >>> y_true = np.array([9.5, 10.5, 12.5])
+    >>> y_intervals = np.array([
+    ... [[9, 9], [10.0, 10.0]],
+    ... [[8.5, 9], [12.5, 12]],
+    ... [[10.5, 10.5], [12.0, 12]]
+    ... ])
+    >>> print(hsic(y_true, y_intervals))
+    [0.31787614 0.2962914 ]
+    """
+    y_true, y_intervals = check_array_shape_regression(y_true, y_intervals)
+    kernel_sizes = cast(NDArray, column_or_1d(kernel_sizes))
+    if len(kernel_sizes) != 2:
+        raise ValueError(
+            "kernel_sizes should be an ArrayLike of length 2"
+        )
```

Comment:
Open question: is there a test that checks whether the ValueError is raised?

Reply:
Yes there is : test_hsic_invalid_kernel_sizes(kernel_sizes=[1])